lib/opFns.js

```diff
@@ -318,6 +318,22 @@ module.exports = {
       cb(err)
     })
   },
+  RETURNDATASIZE: function (runState) {
+    return utils.intToBuffer(runState.lastReturned.length)
+  },
+  RETURNDATACOPY: function (memOffset, returnDataOffset, length, runState) {
+    memOffset = utils.bufferToInt(memOffset)
+    returnDataOffset = utils.bufferToInt(returnDataOffset)
+    length = utils.bufferToInt(length)
+    
+    if (returnDataOffset + length > runState.lastReturned.length) {
+      trap(ERROR.OUT_OF_GAS)
```

Comment:
all EVM exceptions have the same behavior (state changes are reverted and remaining gas is consumed). The error message only determines what gets printed in the EVM trace log, though it would be good to [standardize that](https://github.com/ethereum/tests/issues/249).